http/transport.go

```diff
@@ -0,0 +1,92 @@
+package http
+
+import (
+	"context"
+	"errors"
+	"io"
+	"net"
+	"net/http"
+	"sync"
+	"time"
+
+	"knative.dev/kn-plugin-func/k8s"
+)
+
+type RoundTripCloser interface {
+	http.RoundTripper
+	io.Closer
+}
+
+// NewRoundTripper returns new closable RoundTripper that first tries to dial connection in standard way,
+// if the dial operation fails due to hostname resolution the RoundTripper tries to dial from in cluster pod.
+//
+// This is useful for accessing cluster internal services (pushing a CloudEvent into Knative broker).
+func NewRoundTripper() RoundTripCloser {
+	d := &dialer{
+		netDialer: net.Dialer{
+			Timeout:   30 * time.Second,
+			KeepAlive: 30 * time.Second,
+		},
+	}
+	return &roundTripCloser{
+		Transport: http.Transport{
+			Proxy:                 http.ProxyFromEnvironment,
+			DialContext:           d.DialContext,
+			ForceAttemptHTTP2:     true,
+			MaxIdleConns:          100,
+			IdleConnTimeout:       90 * time.Second,
+			TLSHandshakeTimeout:   10 * time.Second,
+			ExpectContinueTimeout: 1 * time.Second,
```

Comment:
Might be actually great to make those configurable. Not sure how though 😄

Reply:
These are `Go` defaults, and they are not configurable so I think that's OK.